xarray/tests/test_dataset.py

```diff
@@ -1434,6 +1435,7 @@ def test_copy(self):
 
         for copied in [data.copy(deep=False), copy(data)]:
             self.assertDatasetIdentical(data, copied)
+            self.assertEqual(data.encoding, copied.encoding)
```

Comment:
note on behalf of @MaximilianR -

this can just be `assert data.encoding == copied.encoding`

Reply:
Doesn't assertEqual produce better output in case of failure?

Reply:
Pytest satisfies all your desires!

Reply:
@crusaderky - pytest has nice introspection for this sort of thing. https://docs.pytest.org/en/latest/assert.html#making-use-of-context-sensitive-comparisons